app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt

```diff
@@ -0,0 +1,109 @@
+import net.skyscanner.backpack.calendar.BpkCalendar
+import net.skyscanner.backpack.calendar.model.CalendarRange
+import net.skyscanner.backpack.demo.R
+import org.threeten.bp.LocalDate
+
+object DocsRegistry {
+  val screenshots = listOf(
+    Screenshot("Badge", "all"),
+    Screenshot("Button - Primary", "primary"),
+    Screenshot("Button - Secondary", "secondary"),
+    Screenshot("Button - Featured", "featured"),
+    Screenshot("Button - Destructive", "destructive"),
+    Screenshot("Button - Outline", "outline"),
+    Screenshot("ButtonLink - Default", "default"),
+    Screenshot("Calendar - Default", "range", ::setupCalendar),
+    Screenshot("Card - Default", "default"),
+    Screenshot("Card - Without padding", "without-padding"),
+    Screenshot("Card - Selected", "selected"),
+    Screenshot("Card - Corner style large", "corner-style-large"),
+    Screenshot("Card - With divider", "with-divider"),
+    Screenshot("Card - With divider arranged vertically", "with-divider-arranged-vertically"),
+    Screenshot("Card - With divider without padding", "with-divider-without-padding"),
+    Screenshot("Card - With divider and corner style large", "with-divider-and-corner-style-large"),
+    Screenshot("Chip", "all"),
+    Screenshot("Checkbox", "default"),
+    Screenshot("Dialog - With call to action", "with-cta", ::setupDialog),
+    Screenshot("Dialog - Delete confirmation", "delete-confirmation", ::setupDialog),
+    Screenshot("Dialog - Flare", "with-flare", ::setupDialog),
+    Screenshot("Flare - Default", "default"),
+    Screenshot("Flare - Pointer offset", "pointer-offset"),
+    Screenshot("Flare - Rounded", "rounded"),
+    Screenshot("Flare - Inset padding mode", "inset-padding"),
+    Screenshot("Horizontal Nav", "default"),
+    Screenshot("Floating Action Button", "default"),
+    Screenshot("Nav Bar - Default", "expanded"),
+    Screenshot("Nav Bar - Default", "collapsed", ::setupNavBarCollapsed),
+    Screenshot("Nav Bar - With Menu", "navigation", ::setupNavBarCollapsed),
+    Screenshot("Panel", "all"),
+    Screenshot("Rating - Default", "default"),
+    Screenshot("Rating - Sizes", "sizes"),
+    Screenshot("Rating - Size Vertical", "vertical"),
+    Screenshot("Snackbar", "default", ::setupSnackbar),
+    Screenshot("Star Rating - Default", "default"),
+    Screenshot("Star Rating Interactive", "default"),
+    Screenshot("Switch", "default"),
+    Screenshot("Text - Default", "default"),
+    Screenshot("Text - Emphasized", "emphasized"),
+    Screenshot("Text - Heavy", "heavy"),
+    Screenshot("Text Field", "default"),
+    Screenshot("Text Spans", "default"),
+    Screenshot("Spinner - Default", "default"),
+    Screenshot("Spinner - Small", "small"),
+    // Leave toast last as it stays visible in the screen for a while
+    Screenshot("Toast", "default", ::setupToast)
+    ).map { it.toArgs() }
+}
+
+data class Screenshot(
+  val name: String,
+  val screenshotName: String,
+  val setup: (() -> Unit)? = null
```

Comment:
You can replace nullable function with empty function like this:
() -> Unit = {}. Less nullability checks, more safe code :)

At your discretion.

Reply:
I think a nullable value conveys better that this is not a required step and only present for a few use cases. Plus nullables are safe in Kotlin, the compiler will force you to treat them accordingly.